src/test/java/com/epam/tc/hw2/ex1/Exercise1.java

```diff
@@ -0,0 +1,169 @@
+
+        driver.manage().window().maximize();
+        driver.navigate()
+                .to("https://jdi-testing.github.io/jdi-light/index.html");
+        new WebDriverWait(driver, Duration.ofSeconds(10))
+                .until(ExpectedConditions.presenceOfElementLocated(By.id("jdi-frame-site")));
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(driver.getCurrentUrl()).as("Incorrect page opened").isEqualTo("https://jdi-testing.github.io/jdi-light/index.html");
+
+        //2. Assert Browser title
+        String browserTitle = driver.getTitle();
+        softly.assertThat(browserTitle).as("Incorrect browser title").isEqualTo("Home Page");
+
+        //3. Perform login
+        WebElement userIcon = driver.findElement(By.id("user-icon"));
+        userIcon.click();
+        WebElement login = driver.findElement(By.id("name"));
+        login.sendKeys("Roman");
+        WebElement password = driver.findElement(By.id("password"));
+        password.sendKeys("Jdi1234");
+        WebElement enterButton = driver.findElement(By.id("login-button"));
+        enterButton.click();
+
+        //4. Assert Username is loggined
+        WebElement userName = driver.findElement(By.id("user-name"));
+        System.out.println(userName.getText());
+        softly.assertThat(userName.getText()).as("Incorrect user name").isEqualTo("ROMAN IOVLEV");
+
+        //5. Assert that there are 4 items on the header section are displayed and they have proper texts
+        WebElement homeButton = driver.findElement(By.cssSelector("a[href='index.html']"));
+        System.out.println(homeButton.getText());
+        softly.assertThat(homeButton.getText()).as("Incorrect Home button name").isEqualTo("HOME");
+
+        WebElement contactFormButton = driver.findElement(By.cssSelector("a[href='contacts.html']"));
+        softly.assertThat(contactFormButton.getText())
+                .as("Incorrect Contacts button name")
+                .isEqualTo("CONTACT FORM");
+
+        WebElement serviceButton = driver.findElement(By.cssSelector("a.dropdown-toggle"));
+        softly.assertThat(serviceButton.getText()).as("Incorrect Service button name").isEqualTo("SERVICE");
+
+        WebElement metalsColorsButton = driver.findElement(By.cssSelector("a[href='metals-colors.html']"));
+        softly.assertThat(metalsColorsButton.getText())
+                .as("Incorrect METALS & COLORS button name").isEqualTo("METALS & COLORS");
+
+        //6. Assert that there are 4 images on the Index Page and they are displayed
+        List<WebElement> icons = driver.findElements(By.cssSelector(".benefit-icon"));
+
+
+        softly.assertThat(icons.size()).as("Number of images are " + icons.size() + " not 4").isEqualTo(4);
+        Iterator<WebElement> iconsIterator = icons.iterator();
+        while (iconsIterator.hasNext()) {
+            softly.assertThat(iconsIterator.next().isDisplayed()).as("One or more image not displayed").isTrue();
+        }
```

Comment:
It's ok but https://www.w3schools.com/java/java_foreach_loop.asp For-Each loop provides less code(without next)

Reply:
It's not obligatory

Reply:
Fixed